models/requestObject.js

```diff
@@ -59,10 +59,7 @@ function construct (object, additionalData, protocol) {
     this.data.body = object.body;
   }
   else if (object.query !== undefined) {
-    this.data = object;
-  }
-  else if (additionalData.query !== undefined) {
-    this.data = additionalData;
+    this.data.body = {query: object.query};
```

Comment:
I don't remember where the `additionalData` can be used, but wouldn't it be safer to keep the test `else if (additionalData.query...` and simply store the query part of it in `this.data.body`?